From 8c83cb29ed760dca0e9d0516575ab8aa4fcf8dd1 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sun, 05 Aug 2018 09:50:59 +0300
Subject: [PATCH] Add BTL support fixup

Update offsets according to new layout
Use dedicated scratchpad for BTLID
---

diff --git a/board/habanalabs/dali/btl/Makefile b/board/habanalabs/dali/btl/Makefile
index 3090e41..4c30eb7 100644
--- a/board/habanalabs/dali/btl/Makefile
+++ b/board/habanalabs/dali/btl/Makefile
@@ -25,20 +25,20 @@
 
 PRIMARY_CPP_FLAGS += $(COMMON_CPP_FLAGS)
 PRIMARY_CPP_FLAGS += -DBTLID=0x31
-PRIMARY_CPP_FLAGS += -DPCIE_PHY_FW_OFFSET=0x280000
+PRIMARY_CPP_FLAGS += -DPCIE_PHY_FW_OFFSET=0x1b0000
 PRIMARY_CPP_FLAGS += -DPCIE_PHY_FW_SIZE=$(PCIE_PHY_FW_SIZE)
-PRIMARY_CPP_FLAGS += -DSPL_OFFSET=0x40000
+PRIMARY_CPP_FLAGS += -DSPL_OFFSET=0xa0000
 PRIMARY_CPP_FLAGS += -DSPL_SIZE=$(SPL_SIZE)
-PRIMARY_CPP_FLAGS += -DUBOOT_OFFSET=0x60000
+PRIMARY_CPP_FLAGS += -DUBOOT_OFFSET=0xb0000
 PRIMARY_CPP_FLAGS += -DUBOOT_SIZE=$(UBOOT_SIZE)
 
 SECONDARY_CPP_FLAGS += $(COMMON_CPP_FLAGS)
 SECONDARY_CPP_FLAGS += -DBTLID=0x32
-SECONDARY_CPP_FLAGS += -DPCIE_PHY_FW_OFFSET=0x290000
+SECONDARY_CPP_FLAGS += -DPCIE_PHY_FW_OFFSET=0x9b0000
 SECONDARY_CPP_FLAGS += -DPCIE_PHY_FW_SIZE=$(PCIE_PHY_FW_SIZE)
-SECONDARY_CPP_FLAGS += -DSPL_OFFSET=0x50000
+SECONDARY_CPP_FLAGS += -DSPL_OFFSET=0x8a0000
 SECONDARY_CPP_FLAGS += -DSPL_SIZE=$(SPL_SIZE)
-SECONDARY_CPP_FLAGS += -DUBOOT_OFFSET=0x160000
+SECONDARY_CPP_FLAGS += -DUBOOT_OFFSET=0x8b0000
 SECONDARY_CPP_FLAGS += -DUBOOT_SIZE=$(UBOOT_SIZE)
 
 PRSTN_CPP_FLAGS += $(PRIMARY_CPP_FLAGS)
diff --git a/board/habanalabs/dali/btl/primary.S b/board/habanalabs/dali/btl/primary.S
index 04b11e8..2d68d70 100644
--- a/board/habanalabs/dali/btl/primary.S
+++ b/board/habanalabs/dali/btl/primary.S
@@ -38,6 +38,7 @@
  * Using scratchpad registers for
  * passing flash layout to CPU SW
  * ------------------------------
+ * 25 - BTL_ID
  * 26 - PCIE_PHY_FW_OFFSET
  * 27 - PCIE_PHY_FW_SIZE
  * 28 - SPL_OFFSET
@@ -45,6 +46,7 @@
  * 30 - UBOOT_OFFSET
  * 31 - UBOOT_SIZE
  */
+cfg 0xCC4B164, BTLID
 cfg 0xCC4B168, PCIE_PHY_FW_OFFSET
 cfg 0xCC4B16C, PCIE_PHY_FW_SIZE
 cfg 0xCC4B170, SPL_OFFSET
